File: src/data/TestDataProvider.ts
import { Player } from "../types/hytale";
import { PlayerData } from "./PlayerDataProvider";

export class TestDataProvider {
  private tick = 0;

  getPlayerData(player: Player): PlayerData {
    this.tick += 1;

    return {
      rank: "Testeur",
      coins: 1200 + (this.tick % 50),
      level: 5 + (this.tick % 3),
      gameName: "Test Arena",
    };
  }
}

File: src/main/java/com/hytale/scoreboard/data/TestDataProvider.java
package com.hytale.scoreboard.data;

import com.hytale.scoreboard.types.Player;

public class TestDataProvider extends PlayerDataProvider {
  private int tick = 0;

  @Override
  public PlayerData getPlayerData(Player player) {
    tick += 1;

    return new PlayerData(
        "Testeur",
        1200 + (tick % 50),
        5 + (tick % 3),
        "Test Arena");
  }
}
